### ISSUE TITLE
Data re-rendering caused by passed slots in v-for loop

### ISSUE THREAD

**Issue by bart** [vuejs/vue#12407]
### Version
2.6.14

### Reproduction link
[codesandbox.io](https://codesandbox.io/embed/objective-rui-hw0c3?fontsize=14&hidenavigation=1&theme=dark)

### Steps to reproduce
Enter name of a fruit and click "Add fruit". When removing the passed "slot" it is working as expected.

### What is expected?
Neither on typing nor on adding timestamps should change because corresponding data isn't changed.

### What is actually happening?
Timestamps are updating during typing or when adding a new fruit

---
I figured out that the same issue occurs on Vue 3. Any ideas what I'm doing wrong here or how I can pass slots down to looped children without re-rendering everything when updating parent data? You help is highly appreciated, thanks!

**Comment by dannycoll** [comment]
Could you not just pass in a prop to fruit that takes in the date.getTime the fruit was created at?
Doing 
```
  data() {
    return {
      newFruit: "",
      fruits: [
        { id: "1", name: "Apple", date: new Date().getTime() },
        { id: "2", name: "Banana", date: new Date().getTime() },
        { id: "3", name: "Blueberry", date: new Date().getTime() },
      ],
    };
  }
  ```
  and 
  ```
  <template>
  <li>{{ fruit }} ({{ date }})</li>
</template>

<script>
export default {
  name: "Fruit",
  components: {},
  props: ["fruit", "date"],
};
</script>
```
solved the times updating for me

**Comment by bart** [comment]
Using this get time thing is just a way to demonstrate if a part of a component gets re-rendered here. It's not about using this in a real world example. I figured out that in Vue 2 there is no way to solve it. As soon as you use slots whenever data changes the corresponding components get re-rendered. Even if it is keyed correctly. In Vue 3 there is a directive called `v-memo` that might be used to solve it.

I switched to a complete different approach passing components as parameters which then get loaded dynamically via `<component :is=[componentName]>`. Works well and doesn't re-render. So for anyone else facing the same issue trying to improve looped data component rendering this is the way to go.

Cheers!
